crates/router/src/db/merchant_account.rs

```diff
@@ -294,6 +299,47 @@ impl MerchantAccountInterface for Store {
 
         Ok(is_deleted)
     }
+
+    async fn list_multiple_merchant_accounts(
+        &self,
+        merchant_ids: Vec<String>,
+    ) -> CustomResult<Vec<domain::MerchantAccount>, errors::StorageError> {
+        let conn = connection::pg_connection_read(self).await?;
+
+        let encrypted_merchant_accounts =
+            storage::MerchantAccount::list_multiple_merchant_accounts(&conn, merchant_ids)
+                .await
+                .map_err(Into::into)
+                .into_report()?;
+
+        let db_master_key = self.get_master_key().to_vec().into();
+
+        let merchant_key_stores = self
+            .list_multiple_key_stores(
+                encrypted_merchant_accounts
+                    .iter()
+                    .map(|merchant_account| &merchant_account.merchant_id)
+                    .cloned()
+                    .collect(),
+                &db_master_key,
+            )
+            .await?;
+
+        let merchant_accounts = futures::future::try_join_all(
+            encrypted_merchant_accounts
+                .into_iter()
+                .zip(merchant_key_stores.iter())
```

Comment:
How is the order ensured here? what if one of the merchant account or key stores are missing and that disrupts the mapping?